Tests/NetworkingTests/APIManagerTests.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  APIManagerTests.swift
+//
+//
+//  Created by Matej Molnár on 28.12.2023.
+//
+
+@testable import Networking
+import XCTest
+
+@NetworkingActor
+final class APIManagerTests: XCTestCase {
+    enum UserRouter: Requestable {
+        case users(page: Int)
+
+        var baseURL: URL {
+            // swiftlint:disable:next force_unwrapping
+            URL(string: "https://reqres.in/api")!
+        }
+
+        var path: String {
+            switch self {
+            case .users:
+                "users"
+            }
+        }
+
+        var urlParameters: [String: Any]? {
+            switch self {
+            case let .users(page):
+                ["page": page]
+            }
+        }
+
+        var method: HTTPMethod {
+            switch self {
+            case .users:
+                .get
+            }
+        }
+    }
+
+    private let mockSessionId = "2023-01-04T16:15:29Z"
+
+    func testMultiThreadRequests() async throws {
+        let mockResponseProvider = MockResponseProvider(with: Bundle.module, sessionId: mockSessionId)
+        let apiManager = APIManager(
+            responseProvider: mockResponseProvider,
+            // Since one of the mocked responses returns 400 we don't want the test fail.
+            responseProcessors: []
+        )
+
+        // Create 15 parallel requests on multiple threads to test the managers thread safety.
+        try await withThrowingTaskGroup(of: Void.self) { group in
+            for _ in 0..<15 {
+                group.addTask {
+                    try await apiManager.request(UserRouter.users(page: 2))
+                }
+            }
+
+            try await group.waitForAll()
+        }
```

Comment:
Did you also make it crash in the beginning? To verify that it wasn't thread-safe before 😄

Reply:
Yes 😄

Reply:
Cool, thank you 👍

Reply:
I am also trying to implement similar tests for the upload/download managers.